experimental/uefi/baremetal/src/multiboot.rs

```diff
@@ -55,3 +56,59 @@ impl BootInfo<multiboot_mmap_entry> for multiboot_info {
         }
     }
 }
+
+/// Magic number identifying the Multiboot header.
+#[repr(u32)]
+enum Magic {
+    MULTIBOOT1 = 0x1BADB002,
+}
+
+bitflags! {
+    /// Flags specify features requested or required from an boot loader.
+    ///
+    /// See <https://www.gnu.org/software/grub/manual/multiboot/multiboot.html#Header-magic-fields>
+    /// for more details.
+    struct HeaderFlags: u32 {
+        /// Align all boot modules on i386 page (4KB) boundaries.
+        const MULTIBOOT_PAGE_ALIGN = 0x00000001;
+
+        /// Must pass memory information to OS.
+        const MULTIBOOT_MEMORY_INFO = 0x00000002;
+
+        /// Must pass video information to OS.
+        const MULTIBOOT_VIDEO_MODE = 0x00000004;
+
+        /// This flag indicates the use of the address fields in the header.
+        const MULTIBOOT_AOUT_KLUDGE = 0x00010000;
+    }
+}
+
+/// Multiboot header.
+///
+/// The Multiboot header must be contained completely within the first 8192
```

Comment:
nit: this block seems narrower than the max width or 100 chars. also line 102-103.